test/test_dataDict.py

```diff
@@ -0,0 +1,127 @@
+import gw_eccentricity
+from gw_eccentricity import load_data
+from gw_eccentricity import measure_eccentricity
+from gw_eccentricity.utils import time_deriv_4thOrder
+import numpy as np
+
+
+def test_dataDict():
+    """
+    Test that the interface works for dataDict with all allowed
+    combination of keys in dataDict.
+    """
+    # Load test waveform
+    lal_kwargs = {"approximant": "EccentricTD",
+                  "q": 1.0,
+                  "chi1": [0.0, 0.0, 0.0],
+                  "chi2": [0.0, 0.0, 0.0],
+                  "Momega0": 0.01,
+                  "ecc": 0.1,
+                  "mean_ano": 0,
+                  "include_zero_ecc": True}
+    dataDict = load_data.load_waveform(**lal_kwargs)
+
+    # get amplm, phaselm, omegalm
+    phaselm = {}
+    amplm = {}
+    omegalm = {}
+    dt = dataDict["t"][1] - dataDict["t"][0]
+    for k in dataDict["hlm"]:
+        phaselm[k] = -np.unwrap(np.angle(dataDict["hlm"][k]))
+        amplm[k] = np.abs(dataDict["hlm"][k])
+        omegalm[k] = time_deriv_4thOrder(phaselm[k], dt)
+
+    # get amplm_zeroecc, phaselm_zeroecc, omegalm_zeroecc
+    phaselm_zeroecc = {}
+    amplm_zeroecc = {}
+    omegalm_zeroecc = {}
+    dt_zeroecc = dataDict["t_zeroecc"][1] - dataDict["t_zeroecc"][0]
+    for k in dataDict["hlm_zeroecc"]:
+        phaselm_zeroecc[k] = -np.unwrap(np.angle(dataDict["hlm_zeroecc"][k]))
+        amplm_zeroecc[k] = np.abs(dataDict["hlm_zeroecc"][k])
+        omegalm_zeroecc[k] = time_deriv_4thOrder(phaselm_zeroecc[k],
+                                                 dt_zeroecc)
+
+    # List of all available methods
+    available_methods = gw_eccentricity.get_available_methods()
+
+    # create data dict with hlm
+    data_hlm = {"t": dataDict["t"],
+                "hlm": dataDict["hlm"]}
+    # create data dict with amplm and phaselm
+    data_amplm_phaselm = {
+        "t": dataDict["t"],
+        "amplm": amplm,
+        "phaselm": phaselm}
+
+    for method in available_methods:
+        # Using hlm
+        if "Residual" in method:
+            # add zeroecc hlm
+            data_hlm.update({"t_zeroecc": dataDict["t_zeroecc"],
+                             "hlm_zeroecc": dataDict["hlm_zeroecc"]})
```

Comment:
Why not just do this in line 49 itself?

Reply:
The residual data are not required for methods other than Residual ones. In most cases, users may try the non-residual methods first. Thus, I don't include the zeroecc data when the method is not Residual to make sure that there is no hidden dependence on zeroecc data in non-residual methods.

Reply:
But, if I understand correctly, once you do this update, zeroecc data will be used for all methods, including nonresidual ones. Doesn't that defeat the purpose of what you are saying? Unless you get lucky and the order of the methods is so that the Residual ones happen at the end. But then, if someone creates a new method with some random name, that won't be covered in this scheme.

How about the much simpler: 
Make a function that takes two data_dicts: data_dict_hlm and data_dict_other.
This function compares the output of the two, and you just call this function multiple times with different dicts as data_dict_other.
Do the whole exercise with data_dicts with and without zero_ecc data. When the function tries to test it for a Residual method, but the data_dict doesn't contain residual data, just make it `continue` in the loop.
Something like that would make the code much easier to follow, and with far fewer long variable names. 
Added bonus, you test the nonresidual methods with and without zeroecc data, which are both realistic scenarios for a user.
What say?

Reply:
Okay, did change the code. I think it now checks more combinations of keys than I was doing earlier.